ecommerce/api.py

```diff
@@ -814,14 +814,6 @@ def validate_basket_for_checkout(basket):
 
     # Basket item runs must be linked to the basket item product
     run_queryset = product.run_queryset
-    if (
-        CourseRunSelection.objects.filter(basket=basket)
-        .exclude(run__in=run_queryset)
-        .exists()
-    ):
-        raise ValidationError(
-            {"runs": "Some runs present in basket which are not part of product"}
-        )
```

Comment:
I don't think removing this validation is the way to go. You can tweak the queryset a bit to include any eligible courserun, however we do run the risk of changing the price based on whether a different run was shown or not.